scripts/main/settings.js

```diff
@@ -492,46 +492,79 @@ settings.save_enter = function (e) {
 /**
  * @returns {void}
  */
-settings.resetToken = function () {
-	api.post(
-		"User::resetToken",
-		{},
-		/**
-		 *
-		 * @param {User} data
-		 */
-		function (data) {
-			let bodyHtml = lychee.html`<div class='directLinks'><p><span id="apiToken">${
-				data.token
-			}</span> <a id="button_copy_token" class='basicModal__button' title='${lychee.locale["URL_COPY_TO_CLIPBOARD"]}'>${build.iconic(
-				"copy",
-				"ionicons"
-			)}</a> <a id="button_disable_token" class='basicModal__button' title='${lychee.locale["DISABLE"]}'>${build.iconic("ban")}</a></p></div>`;
-			basicModal.show({
-				body: bodyHtml,
-				buttons: {
-					action: {
-						title: lychee.locale["RESET"],
-						fn: function () {
-							basicModal.close();
-							settings.resetToken();
-						},
-						class: "red",
-					},
-					cancel: {
-						title: lychee.locale["CLOSE"],
-						fn: basicModal.close,
-					},
-				},
-			});
-			$("#button_copy_token").on(lychee.getEventName(), function () {
-				navigator.clipboard.writeText(data.token);
-			});
-			$("#button_disable_token").on(lychee.getEventName(), function () {
-				api.post("User::unsetToken", {}, function () {
-					$("#apiToken").html("disabled");
-				});
-			});
+settings.openTokenDialog = function () {
+	let token = "";
+
+	/**
+	 * @param {boolean} has
+	 * @param {boolean} canBeViewed
+	 */
+	const updateBtnVisibility = function (has, canBeViewed) {
```

Comment:
I'm not quite happy with the parameters.

`has` is to unspecific for my taste. I know what you mean from context, it means `hasToken` but I doubt that anybody does remember that in half a year. Moreover, the parameter is actually not necessary. The method can simply use `lychee.user.has_token` internally. No need, to pass that as an argument at all.

`canBeViewed` should rather be named like `isTokenValueVisible`. Again, in half a year nobody remembers _what_ can be viewed if the parameter is only called like like that.

I will change it.

Reply:
Fixed in https://github.com/LycheeOrg/Lychee-front/pull/300/commits/bca3d5558ddcabeb52334f6397f425b1a549636d. IMHO this can be resolved.